packages/lodestar/src/network/encoder.ts

```diff
@@ -0,0 +1,131 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {ERR_INVALID_REQ, MAX_CHUNK_SIZE, Method, ReqRespEncoding} from "../constants";
+import {RequestBody, ResponseBody} from "@chainsafe/lodestar-types";
+import {IReqRespEncoder} from "./encoders/interface";
+import {getRequestMethodSSZType, getResponseMethodSSZType} from "./util";
+import {SszEncoder} from "./encoders/ssz";
+import {SnappyEncoder} from "./encoders/snappy";
+import * as varint from "varint";
+import {Response, ResponseChunk} from "./interface";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import BufferList from "bl";
```

Comment:
Why are we using BufferList?

Reply:
Because that is actual type returned from source. Check: https://github.com/jacobheun/it-handshake#handshakeduplex
We were using source.slice() which is method on BufferList to return Buffer.

Reply:
I've added bl types as dev dependency but it doesn't let me import just type